orchestra/models.py

```diff
@@ -523,3 +525,48 @@ class PayRate(PayRateMixin, models.Model):
     hourly_multiplier = models.DecimalField(max_digits=6, decimal_places=4)
     start_date = models.DateField(default=timezone.now)
     end_date = models.DateField(null=True, blank=True)
+
+
+class CommunicationType(BaseModel):
+    """
+    A CommunicationType object defines a certain type of communication with
+    Users.
+
+    Attributes:
+        slug (str):
+            Unique identifier for the communication type.
+        description (str):
+            A longer description of the communication type.
+    """
+    slug = models.CharField(max_length=100)
+    description = models.CharField(max_length=255)
+
+    class Meta:
+        app_label = 'orchestra'
+
+
+class CommunicationPreference(models.Model):
+    """
+    A CommunicationPreference object defines how a User would like to contacted
+    for a given CommunicationType.
+
+    Attributes:
+        user (django.contrib.auth.models.User):
+            Django user whom the preference represents.
+        methods (ChoicesEnum):
+            The ways in which the user would like to be contacted.
+        type (CommunicationType):
+            The type of communication to which this preference applies.
+    """
+
+    class CommunicationMethods(ChoicesEnum):
+        SLACK = 'slack'
+        EMAIL = 'email'
+
+    user = models.ForeignKey(User)
+    methods = MultiSelectField(
+        choices=CommunicationMethods.choices(), blank=True)
+    type = models.ForeignKey(CommunicationType)
```

Comment:
i feel like this should also be a MultiSelectField/BitField of CommunicationType, where CommunicationType is an enum:

- multi/bit because you can imagine wanting to be alerted in multiple ways
- enum for the same reason methods are enums---we'll want to talk about them programatically. this is assuming I understand the field: it's things like staffbot, etc?